src/test/basics/PerfLog_test.cpp

```diff
@@ -288,23 +289,45 @@ class PerfLog_test : public beast::unit_test::suite
                 perms::remove_perms | perms::owner_write |
                 perms::others_write | perms::group_write);
 
-            // Now construct a PerfLog.  The PerfLog should attempt to shut
-            // down the server because it can't open its file.
-            BEAST_EXPECT(parent.stopSignaled == false);
-            auto perfLog {getPerfLog (parent, WithFile::yes)};
-            BEAST_EXPECT(parent.stopSignaled == true);
+            // If the test is running as root, then the write protect may have
+            // no effect.  Make sure write protect worked before proceeding.
+            bool const writeProtected {[&fullPath] ()
+            {
+                std::ofstream verifyWriteProtect {
+                    fullPath.c_str(), std::ios::out | std::ios::app};
 
-            // Start PerfLog and wait long enough for PerfLog::report()
-            // to not be able to write to its file.  That should cause no
-            // problems.
-            parent.doStart();
-            std::this_thread::sleep_for (parent.getLogInterval() * 3);
-            parent.doStop();
+                if (verifyWriteProtect)
+                {
+                    verifyWriteProtect.close();
+                    return false;
+                }
+                return true;
+            }()};
 
-            // Fix file permissions so the file can be cleaned up.
-            permissions (fullPath,
-                perms::add_perms | perms::owner_write |
-                perms::others_write | perms::group_write);
+            if (writeProtected)
+            {
+                // Now construct a PerfLog.  The PerfLog should attempt to shut
+                // down the server because it can't open its file.
+                BEAST_EXPECT(parent.stopSignaled == false);
+                auto perfLog {getPerfLog (parent, WithFile::yes)};
+                BEAST_EXPECT(parent.stopSignaled == true);
+
+                // Start PerfLog and wait long enough for PerfLog::report()
+                // to not be able to write to its file.  That should cause no
+                // problems.
+                parent.doStart();
+                std::this_thread::sleep_for (parent.getLogInterval() * 3);
+                parent.doStop();
+
+                // Fix file permissions so the file can be cleaned up.
+                permissions (fullPath,
+                    perms::add_perms | perms::owner_write |
+                    perms::others_write | perms::group_write);
+            }
+            else
+            {
+                j_.fatal() << "Unable to write protect file.  Test skipped.";
```

Comment:
this feel like something that belongs in a `log()` message since it's test related, but it's not a big deal either way. If you stick with the journal, should it just be a warning?

Reply:
I'm writing to the journal that `Env` provides.  It I write at `fatal` level then it shows up on the console.  I tried writing at `warn` level and the write didn't get to the console.  I felt like getting the message (that we're skipping a test and why) out to the console was a good thing.  But I could be persuaded otherwise.

Reply:
the `j_` journal here is what I would call the "application log"..that's as opposed to the `log` stream that the test suite provides. This is really splitting hairs...I realize you just want to get the message out and I agree that's really what matters. Just for completeness, here's the output using `j_`:

```
ripple.basics.PerfLog
FTL:PerfLog_test Unable to create performance log directory "/tmp/perf_log_test_dir": File exists
FTL:PerfLog_test Unable to write protect file.  Test skipped.
627ms, 1 suite, 1 case, 45636 tests total, 0 failures
```

and if you instead use the test suite log stream like: `log << "Unable to write protect file.  Test skipped." << std::endl;`, then you get:

```
ripple.basics.PerfLog
FTL:PerfLog_test Unable to create performance log directory "/tmp/perf_log_test_dir": File exists
Unable to write protect file.  Test skipped.
725ms, 1 suite, 1 case, 45636 tests total, 0 failures
```

I've beaten this one to death ... and I'm perfectly fine with it either way.